pragma-node/src/infra/repositories/onchain_repository/history.rs

```diff
@@ -70,3 +103,175 @@ pub async fn get_historical_aggregated_entries(
 
     Ok(raw_entries)
 }
+
+/// Retry to get the onchain historical entries by finding
+/// an alternative route.
+/// TODO: This code is very similar to the one in [entry_repository] ;
+///       once we have proper E2E tests, we should try to merge the code.
+pub async fn retry_with_routing(
+    onchain_pool: &Pool,
+    offchain_pool: &Pool,
+    network: &Network,
+    pair_id: String,
+    timestamp_range: &TimestampRange,
+    chunk_interval: &Interval,
+) -> Result<(Vec<HistoricalEntryRaw>, u32), InfraError> {
+    let [base, quote]: [&str; 2] = pair_id
```

Comment:
don't we have an util function for this ? maybe it's time we create an actual `Pair` struct wdyt ?

Reply:
mmmmh agreed, but I also think that it's definitely gonna involve a lot of changes in every handlers - should be probably done in a follow-up PR?

Reply:
after the critical tests (#114 ) probably?

Reply:
(Updated the code so it uses the util function)

Reply:
yup let's create the issue tho

Reply:
Done here: #119